panic: runtime error: index out of range ( 0.6.7 )

Hi thanks for the arm64 version

just made a quick try on a Qnap x86_64 

i have panic error

`[/share/Public] # annie https://youtu.be/DNaOZovrSVo
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/iawia002/annie/extractors/youtube.extractVideoURLS(0xc420474080, 0x1, 0x1, 0x7fff469a8b76, 0x1c, 0xc4203c65a0, 0x29, 0x2)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/extractors/youtube/youtube.go:145 +0xdc5
github.com/iawia002/annie/extractors/youtube.youtubeDownload(0x7fff469a8b76, 0x1c, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/extractors/youtube/youtube.go:105 +0x306
github.com/iawia002/annie/extractors/youtube.Download(0x7fff469a8b76, 0x1c)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/extractors/youtube/youtube.go:65 +0x304
main.download(0x7fff469a8b76, 0x1c)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/main.go:61 +0x42a
main.main()
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/main.go:102 +0x81
`

Do i missed somethin ?

--- a/extractors/youtube/youtube.go
+++ b/extractors/youtube/youtube.go
@@ -14,9 +14,10 @@ import (
 )
 
 type args struct {
-	Title string `json:"title"`
-	// Stream string `json:"url_encoded_fmt_stream_map"`
+	Title  string `json:"title"`
 	Stream string `json:"adaptive_fmts"`
+	// not every page has `adaptive_fmts` field https://youtu.be/DNaOZovrSVo
+	Stream2 string `json:"url_encoded_fmt_stream_map"`
 }
 
 type assets struct {
@@ -100,9 +101,8 @@ func youtubeDownload(uri string) downloader.VideoData {
 	var youtube youtubeData
 	json.Unmarshal([]byte(ytplayer), &youtube)
 	title := youtube.Args.Title
-	streams := strings.Split(youtube.Args.Stream, ",")
 
-	format := extractVideoURLS(streams, uri, youtube.Assets.JS)
+	format := extractVideoURLS(youtube, uri)
 
 	extractedData := downloader.VideoData{
 		Site:    "YouTube youtube.com",
@@ -114,7 +114,11 @@ func youtubeDownload(uri string) downloader.VideoData {
 	return extractedData
 }
 
-func extractVideoURLS(streams []string, referer, assest string) map[string]downloader.FormatData {
+func extractVideoURLS(data youtubeData, referer string) map[string]downloader.FormatData {
+	streams := strings.Split(data.Args.Stream, ",")
+	if data.Args.Stream == "" {
+		streams = strings.Split(data.Args.Stream2, ",")
+	}
 	var ext string
 	var audio downloader.URLData
 	format := map[string]downloader.FormatData{}
@@ -133,6 +137,9 @@ func extractVideoURLS(streams []string, referer, assest string) map[string]downl
 		}
 
 		quality := stream.Get("quality_label")
+		if quality == "" {
+			quality = stream.Get("quality") // for url_encoded_fmt_stream_map
+		}
 		if quality != "" {
 			quality = fmt.Sprintf("%s %s", quality, streamType)
 		} else {
@@ -144,7 +151,10 @@ func extractVideoURLS(streams []string, referer, assest string) map[string]downl
 		} else {
 			ext = utils.MatchOneOf(streamType, `(\w+)/(\w+);`)[2]
 		}
-		realURL := genSignedURL(stream.Get("url"), stream, assest) + "&ratebypass=yes"
+		realURL := genSignedURL(stream.Get("url"), stream, data.Assets.JS)
+		if !strings.Contains(realURL, "ratebypass=yes") {
+			realURL += "&ratebypass=yes"
+		}
 		size := request.Size(realURL, referer)
 		urlData := downloader.URLData{
 			URL:  realURL,
@@ -164,6 +174,10 @@ func extractVideoURLS(streams []string, referer, assest string) map[string]downl
 
 	format["default"] = format[bestQualityItag]
 	delete(format, bestQualityItag)
+
+	if data.Args.Stream == "" {
+		return format
+	}
 
 	// Unlike `url_encoded_fmt_stream_map`, all videos in `adaptive_fmts` have no sound,
 	// we need download video and audio both and then merge them.

--- a/extractors/youtube/youtube_test.go
+++ b/extractors/youtube/youtube_test.go
@@ -51,6 +51,15 @@ func TestYoutube(t *testing.T) {
 			},
 			playlist: true,
 		},
+		{
+			name: "url_encoded_fmt_stream_map test",
+			args: test.Args{
+				URL:     "https://youtu.be/DNaOZovrSVo",
+				Title:   "QNAP Success Story - Scorptec",
+				Size:    16839256,
+				Quality: `hd720 video/mp4; codecs="avc1.64001F, mp4a.40.2"`,
+			},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

